web/src/components/product/ProductPage.jsx

```diff
@@ -0,0 +1,435 @@
+            selectedProduct.name
+          )
+        }
+      </p>
+      <Popup.Actions>
+        <Popup.Confirm onClick={onAccept}>
+          {_("Accept")}
+        </Popup.Confirm>
+      </Popup.Actions>
+    </Popup>
+  );
+};
+
+/**
+ * Buttons for a product that does not require registration.
+ * @component
+ *
+ * @param {object} props
+ * @param {boolean} props.isDisabled
+ */
+const WithoutRegistrationButtons = ({ isDisabled = false }) => {
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+
+  return (
+    <>
+      <Button
+        variant="primary"
+        onClick={() => setIsPopupOpen(true)}
+        isDisabled={isDisabled}
+      >
+        {_("Change product")}
+      </Button>
+      <ProductSelectionPopup
+        isOpen={isPopupOpen}
+        onFinish={() => setIsPopupOpen(false)}
+        onCancel={() => setIsPopupOpen(false)}
+      />
+    </>
+  );
+};
+
+/**
+ * Buttons for a product that is not registered yet.
+ * @component
+ *
+ * @param {object} props
+ * @param {boolean} props.isDisabled
+ */
+const DeregisteredButtons = ({ isDisabled = false }) => {
+  const [isRegistrationPopupOpen, setIsRegistrationPopupOpen] = useState(false);
+  const [isSelectionPopupOpen, setIsSelectionPopupOpen] = useState(false);
+
+  return (
+    <>
+      <Button
+        variant="primary"
+        onClick={() => setIsRegistrationPopupOpen(true)}
+        isDisabled={isDisabled}
+      >
+        {_("Register")}
+      </Button>
+      <Button
+        variant="secondary"
+        onClick={() => setIsSelectionPopupOpen(true)}
+        isDisabled={isDisabled}
+      >
+        {_("Change product")}
+      </Button>
+      <ProductRegistrationPopup
+        isOpen={isRegistrationPopupOpen}
+        onFinish={() => setIsRegistrationPopupOpen(false)}
+        onCancel={() => setIsRegistrationPopupOpen(false)}
+      />
+      <ProductSelectionPopup
+        isOpen={isSelectionPopupOpen}
+        onFinish={() => setIsSelectionPopupOpen(false)}
+        onCancel={() => setIsSelectionPopupOpen(false)}
+      />
+    </>
+  );
+};
+
+/**
+ * Buttons for a product that is already registered.
+ * @component
+ *
+ * @param {object} props
+ * @param {boolean} props.isDisabled
+ */
+const RegisteredButtons = ({ isDisabled = false }) => {
+  const [isDeregistrationPopupOpen, setIsDeregistrationPopupOpen] = useState(false);
+  const [isWarningPopupOpen, setIsWarningPopupOpen] = useState(false);
+
+  return (
+    <>
+      <Button
+        variant="primary"
+        onClick={() => setIsDeregistrationPopupOpen(true)}
+        isDisabled={isDisabled}
+      >
+        {_("Deregister")}
+      </Button>
+      <Button
+        variant="secondary"
+        onClick={() => setIsWarningPopupOpen(true)}
+        isDisabled={isDisabled}
+      >
+        {_("Change product")}
+      </Button>
+      <ProductDeregistrationPopup
+        isOpen={isDeregistrationPopupOpen}
+        onFinish={() => setIsDeregistrationPopupOpen(false)}
```

Comment:
The same as in other comment. I think it is better creating these functions just once, before the return.